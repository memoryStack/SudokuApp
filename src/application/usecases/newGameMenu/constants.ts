// TODO: how to compose enums so that's it's easier to edit or re-organize them
// TODO: these names are confusing, fix these
export enum AUTO_GENERATED_NEW_GAME_IDS {
    EASY = 'EASY',
    MEDIUM = 'MEDIUM',
    HARD = 'HARD',
    EXPERT = 'EXPERT',
}

export enum NEW_GAME_IDS {
    EASY = 'EASY',
    MEDIUM = 'MEDIUM',
    HARD = 'HARD',
    EXPERT = 'EXPERT',
    CUSTOMIZE_PUZZLE = 'CUSTOMIZE_PUZZLE',
}

export enum START_GAME_MENU_ITEMS_IDS {
    EASY = 'EASY',
    MEDIUM = 'MEDIUM',
    HARD = 'HARD',
    EXPERT = 'EXPERT',
    CUSTOMIZE_PUZZLE = 'CUSTOMIZE_PUZZLE',
    RESUME = 'RESUME'
}
