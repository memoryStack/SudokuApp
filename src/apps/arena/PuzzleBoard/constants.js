export const SMART_HINT_TRY_OUT_ACTION_PROP_NAME = 'smartHintTryOutOnAction'