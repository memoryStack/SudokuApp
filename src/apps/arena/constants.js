import { roundToNearestPixel } from '../../utils/util'

export const BOARD_AXES_VALUES = {
    Y_AXIS: ['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I'],
    X_AXIS: ['1', '2', '3', '4', '5', '6', '7', '8', '9'],
}

export const BOARD_CELLS_COUNT = 81
export const MAX_INSTANCES_OF_NUMBER = 9
export const CELLS_IN_HOUSE = 9
export const NUMBERS_IN_HOUSE = 9
export const HOUSES_COUNT = 9
export const BOARD_TO_WINDOW_WIDTH_RATIO = 0.94
export const THICK_BORDER_COUNT = 2
export const THIN_BORDER_COUNT = 8
export const BLOCKS_COUNT_IN_ROW = 3

export const GRID_TRAVERSALS = {
    ROW: 'ROW',
    COL: 'COL',
    BLOCK: 'BLOCK',
}

export const STATIC_BOARD_ELEMENTS_DIMENSIONS = {
    THICK_BORDER_WIDTH: roundToNearestPixel(3),
    THIN_BORDER_WIDTH: roundToNearestPixel(1),
    AXIS_WIDTH: roundToNearestPixel(20),
}

export const DEEPLINK_PUZZLE_URL_ERRORS = {
    EMPTY_URL: 'Invaild Url',
    INCOMPLETE_NUMBERS: `Puzzle doesn't have ${BOARD_CELLS_COUNT} numbers`,
    INVALID_CHARACTERS: 'Puzzle has invalid characters other than 0-9',
}

export const PUZZLE_SOLUTION_TYPES = {
    NO_SOLUTION: 'NO_SOLUTION',
    UNIQUE_SOLUTION: 'UNIQUE_SOLUTION',
    MULTIPLE_SOLUTIONS: 'MULTIPLE_SOLUTIONS',
}

export const BOARD_MOVES_TYPES = {
    ADD: 'ADD',
    REMOVE: 'REMOVE',
}

export const BOARD_GRID_BORDERS_DIRECTION = {
    VERTICAL: 'vertical',
    HORIZONTAL: 'horizontal',
}

export const ARENA_PAGE_TEST_ID = 'arena_test_id'

export const GAME_OVER_CARD_OVERLAY_TEST_ID = 'GAME_OVER_CARD_OVERLAY_TEST_ID'
