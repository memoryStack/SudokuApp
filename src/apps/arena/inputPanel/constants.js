const ACTION_TYPES = {
    ON_ERASE_CLICK: 'ON_ERASE_CLICK',
    ON_NUMBER_CLICK: 'ON_NUMBER_CLICK',
}
export const INPUT_PANEL_CONTAINER_TEST_ID = 'INPUT_PANEL_CONTAINER_TEST_ID'

export const INPUT_PANEL_ITEM_TEST_ID = 'INPUT_PANEL_ITEM_TEST_ID'

export { ACTION_TYPES }
