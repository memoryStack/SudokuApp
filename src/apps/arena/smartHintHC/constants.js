export const FOOTER_BUTTONS_TEXT = {
    NEXT: 'Next',
    PREV: 'Prev',
    APPLY_HINT: 'Apply Hint',
}

export const SMART_HINT_CHANGES_APPLY_DELAY = 100
