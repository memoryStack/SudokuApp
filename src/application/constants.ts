import { AUTO_GENERATED_NEW_GAME_IDS } from "./usecases/newGameMenu/constants"

export enum BOARD_MOVES_TYPES {
    ADD = 'ADD',
    REMOVE = 'REMOVE',
}

export enum PENCIL_STATE {
    ACTIVE = 'ACTIVE',
    INACTIVE = 'INACTIVE',
}

export enum GAME_STATE {
    ACTIVE = 'ACTIVE',
    INACTIVE = 'INACTIVE',
    DISPLAY_HINT = 'DISPLAY_HINT',
    GAME_SELECT = 'GAME_SELECT',
    OVER_SOLVED = 'SOLVED',
    OVER_UNSOLVED = 'UNSOLVED',
}

export enum LEVEL_DIFFICULTIES {
    EASY = 'EASY',
    MEDIUM = 'MEDIUM',
    HARD = 'HARD',
    EXPERT = 'EXPERT',
}

export const GAME_LEVEL_VS_CLUES = {
    [AUTO_GENERATED_NEW_GAME_IDS.EASY]: 36,
    [AUTO_GENERATED_NEW_GAME_IDS.MEDIUM]: 32,
    [AUTO_GENERATED_NEW_GAME_IDS.HARD]: 28,
    [AUTO_GENERATED_NEW_GAME_IDS.EXPERT]: 24,
}
