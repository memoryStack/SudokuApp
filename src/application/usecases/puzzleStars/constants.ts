import { NEW_GAME_IDS } from "../newGameMenu/constants"

const TIME_LIMIT_FOR_ONE_STAR = 100000000

// this timeLimit is in minutes
export const PUZZLE_STARS_DISTRIBUTION = {
    [NEW_GAME_IDS.EASY]: [{ stars: 3, timeLimit: 20 }, { stars: 2, timeLimit: 30 }, { stars: 1, timeLimit: TIME_LIMIT_FOR_ONE_STAR }],
    [NEW_GAME_IDS.MEDIUM]: [{ stars: 3, timeLimit: 30 }, { stars: 2, timeLimit: 40 }, { stars: 1, timeLimit: TIME_LIMIT_FOR_ONE_STAR }],
    [NEW_GAME_IDS.HARD]: [{ stars: 3, timeLimit: 40 }, { stars: 2, timeLimit: 50 }, { stars: 1, timeLimit: TIME_LIMIT_FOR_ONE_STAR }],
    [NEW_GAME_IDS.EXPERT]: [{ stars: 3, timeLimit: 50 }, { stars: 2, timeLimit: 60 }, { stars: 1, timeLimit: TIME_LIMIT_FOR_ONE_STAR }]
}
