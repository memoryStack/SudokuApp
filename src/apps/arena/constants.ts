import { BOARD_CELLS_COUNT } from '@domain/board/board.constants'

import { roundToNearestPixel } from '../../utils/util'

export const BOARD_AXES_VALUES = {
    Y_AXIS: ['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I'],
    X_AXIS: ['1', '2', '3', '4', '5', '6', '7', '8', '9'],
}

export const BOARD_TO_WINDOW_WIDTH_RATIO = 0.94
export const THICK_BORDER_COUNT = 2
export const THIN_BORDER_COUNT = 6
export const BLOCKS_COUNT_IN_ROW = 3

export enum GRID_TRAVERSALS {
    ROW = 'ROW',
    COL = 'COL',
    BLOCK = 'BLOCK',
}

export const STATIC_BOARD_ELEMENTS_DIMENSIONS = {
    THICK_BORDER_WIDTH: roundToNearestPixel(3),
    THIN_BORDER_WIDTH: roundToNearestPixel(1),
    AXIS_WIDTH: roundToNearestPixel(20),
    BOARD_BORDER_WIDTH: roundToNearestPixel(1),
}

export const DEEPLINK_PUZZLE_URL_ERRORS = {
    EMPTY_URL: 'Invaild Url',
    INCOMPLETE_NUMBERS: `Puzzle doesn't have ${BOARD_CELLS_COUNT} numbers`,
    INVALID_CHARACTERS: 'Puzzle has invalid characters other than 0-9',
}

export enum PUZZLE_SOLUTION_TYPES {
    NO_SOLUTION = 'NO_SOLUTION',
    UNIQUE_SOLUTION = 'UNIQUE_SOLUTION',
    MULTIPLE_SOLUTIONS = 'MULTIPLE_SOLUTIONS',
}

export const BOARD_MOVES_TYPES = {
    ADD: 'ADD',
    REMOVE: 'REMOVE',
}

export const BOARD_GRID_BORDERS_DIRECTION = {
    VERTICAL: 'vertical',
    HORIZONTAL: 'horizontal',
}

export type BoardGridBorderDirectionValue = typeof BOARD_GRID_BORDERS_DIRECTION[keyof typeof BOARD_GRID_BORDERS_DIRECTION];

export const ARENA_PAGE_TEST_ID = 'arena_test_id'

export const GAME_OVER_CARD_OVERLAY_TEST_ID = 'GAME_OVER_CARD_OVERLAY_TEST_ID'

export const SMART_HEIGHT_HC_MAX_HEIGHT = 250
