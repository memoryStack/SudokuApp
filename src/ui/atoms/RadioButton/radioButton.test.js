import * as React from 'react'

// TODO: manage these imports properly
import { render, screen } from 'testing-utils'
import { PersistentRender } from '@utils/testing/persistentRender'

import Radio from './RadioButton'
import { TEST_IDS } from './radioButton.constants'

// TODO: i don't see options available to me for "screen" object
// this is important to fix

test('snapshot for not selected radio button', () => {
    const tree = render(<Radio isSelected={false} />).toJSON()
    expect(tree).toMatchSnapshot()
})

test('snapshot for selected radio button', () => {
    const tree = render(<Radio isSelected />).toJSON()
    expect(tree).toMatchSnapshot()
})

test('snapshot for disabled radio button', () => {
    const tree = render(<Radio disabled />).toJSON()
    expect(tree).toMatchSnapshot()
})

describe('radio button functionality', () => {
    test('renders only the outer circle if not selected', () => {
        render(<Radio isSelected={false} />)

        expect(screen.getByTestId(TEST_IDS.OUTER_RING)).toBeTruthy()
        expect(screen.queryByTestId(TEST_IDS.INNER_DOT)).toBeFalsy()
    })

    test('renders only inner circle once state changes from not selected to selected', () => {
        render(<Radio isSelected={false} />)

        expect(screen.queryByTestId(TEST_IDS.INNER_DOT)).toBeFalsy()

        screen.update(<Radio isSelected />)

        expect(screen.getByTestId(TEST_IDS.INNER_DOT)).toBeTruthy()
    })

    test('inner dot is not rendered if disabled is true but parent component mistakenly marks Radio as selected', () => {
        render(<Radio isSelected={false} disabled />)

        expect(screen.queryByTestId(TEST_IDS.INNER_DOT)).toBeFalsy()

        screen.update(<Radio isSelected disabled />)

        expect(screen.queryByTestId(TEST_IDS.INNER_DOT)).toBeFalsy()
    })

    test('changes color when selected/disabled state is toggled', () => {
        const aPersistentRender = new PersistentRender(
            {
                getRenderingResultToCompare: () => {
                    const element = screen.getByTestId(TEST_IDS.OUTER_RING)
                    return [element.props.style.borderColor]
                },
            },
        )

        aPersistentRender.render(<Radio isSelected={false} />)
        aPersistentRender.update(<Radio isSelected />)

        expect(aPersistentRender.getChangedRenderingResultStatus()).toEqual([true])

        aPersistentRender.update(<Radio isSelected={false} disabled />)
        expect(aPersistentRender.getChangedRenderingResultStatus()).toEqual([true])
    })
})
