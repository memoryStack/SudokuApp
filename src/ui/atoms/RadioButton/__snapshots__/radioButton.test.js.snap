// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot for disabled radio button 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderColor": "rgba(28, 24, 28, 0.38)",
      "borderRadius": 1000,
      "borderWidth": 2,
      "height": 20,
      "justifyContent": "center",
      "width": 20,
    }
  }
  testID="radioOuterRing"
/>
`;

exports[`snapshot for not selected radio button 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#4D4752",
      "borderRadius": 1000,
      "borderWidth": 2,
      "height": 20,
      "justifyContent": "center",
      "width": 20,
    }
  }
  testID="radioOuterRing"
/>
`;

exports[`snapshot for selected radio button 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#4F2AA2",
      "borderRadius": 1000,
      "borderWidth": 2,
      "height": 20,
      "justifyContent": "center",
      "width": 20,
    }
  }
  testID="radioOuterRing"
>
  <View
    style={
      Object {
        "backgroundColor": "#4F2AA2",
        "borderRadius": 1000,
        "height": 10,
        "width": 10,
      }
    }
    testID="radioInnerDot"
  />
</View>
`;
